### PR TITLE
PUBDEV-8849: Enable GAM I-spline to support increasing and decreasing functions

### PR DESCRIPTION
**Pull request by wendycwong**
This PR completes the task in JIRA: https://h2oai.atlassian.net/browse/PUBDEV-8849.

Add coefficient constraints for I-spline to allow coefficients to be zero or negative.  Add new parameter
ispline_non_negative as an array which is the same size as gam_columns.  If true, coefficients will be non-negative, false, coefficients will be non-positive.

Added Java/Python/R tests.

### CODE REVIEW

File: h2o-py/h2o/estimators/gam.py [hunk 1]
```diff
@@ -858,6 +865,22 @@ def remove_collinear_columns(self, remove_collinear_columns):
         assert_is_type(remove_collinear_columns, None, bool)
         self._parms["remove_collinear_columns"] = remove_collinear_columns
 
+    @property
+    def splines_non_negative(self):
+        """
+        Valid for I-spline only.  True if the I-splines are non-decreasing and False if the I-splines are non-
+        increasing.  If specified, must be the same size as gam_columns.  Values for bs != 2 will be ignored.  Default
```
Comment (michalkurka) [inline h2o-py/h2o/estimators/gam.py:872]: What does `values for bs != 2 will be ignored` mean?

Can we reword it to something like, "this parameter is only valid for monotone splines (bs = 2)`?

**Review by michalkurka — APPROVED** [review]

File: h2o-py/h2o/estimators/gam.py (hunk 1 above)
Comment (wendycwong) [reply h2o-py/h2o/estimators/gam.py]: It meant that bs spline is not i-spline.

Comment (wendycwong) [reply h2o-py/h2o/estimators/gam.py]: Good point.  I will add your suggestion.
